addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2998,6 +3006,14 @@ const SnippetOptionWidget = Widget.extend({
      * @returns {Promise|undefined}
      */
     selectStyle: async function (previewMode, widgetValue, params) {
+        if (this.$target[0].classList.contains('o_we_bg_filter')) {
+            if (params['cssProperty'] === 'background-color' && widgetValue.includes('-gradient(')) {
+                params['cssProperty'] = 'background-image';
+                this.$target[0].style['background-color'] = '';
+            } else {
+                this.$target[0].style['background-image'] = '';
+            }
+        }
```

Comment:
You are editing the generic `selectStyle` method with a "if is bg filter", is this needed? :grimacing: Also, knowing if it is a gradient or not should really be something you get from the `params` variable (not guessing it from the `widgetValue`).